### ISSUE TITLE
ktouch graphical glitches

### ISSUE THREAD

**Issue by davidak** [NixOS/nixpkgs#89430]
**Describe the bug**
ktouch has graphical glitches. at least on my system

can you reproduce it?

terminal output:

```
(ktouch:27904): dbind-WARNING **: 21:09:19.541: Couldn't connect to accessibility bus: Failed to connect to socket /tmp/dbus-JTQXBdtPJW: Connection refused
qCoreApp Application(0x7ffe10c6e0f0)
Fetched layout groups from X server: 	layouts: ("de", "us") 	variants: ("", "")
Fetched layout groups from X server: 	layouts: ("de", "us") 	variants: ("", "")
qrc:/ktouch/qml/main.qml:129:9: QML HomeScreen: Binding loop detected for property "activeKeyboardLayoutName"
Fetched layout groups from X server: 	layouts: ("de", "us") 	variants: ("", "")
Fetched layout groups from X server: 	layouts: ("de", "us") 	variants: ("", "")
qrc:/ktouch/qml/keyboard/KeyItem.qml:235: TypeError: Cannot read property 'width' of null
qrc:/ktouch/qml/keyboard/KeyItem.qml:237: TypeError: Cannot read property 'width' of null
qrc:/ktouch/qml/keyboard/KeyLabel.qml:63: TypeError: Type error
qrc:/ktouch/qml/keyboard/KeyLabel.qml:63: TypeError: Type error
qrc:/ktouch/qml/keyboard/KeyLabel.qml:63: TypeError: Type error
qrc:/ktouch/qml/keyboard/KeyLabel.qml:63: TypeError: Type error
qrc:/ktouch/qml/keyboard/KeyItem.qml:204: TypeError: Cannot read property 'height' of null
qrc:/ktouch/qml/homescreen/CourseSelector.qml:69: TypeError: Type error
```

**To Reproduce**
1. `nix run nixpkgs.ktouch -c ktouch`

**Expected behavior**
no graphical glitches

**Screenshots**
![Screenshot from 2020-06-03 21-10-18](https://user-images.githubusercontent.com/91113/83678905-ca164000-a5de-11ea-9549-a4a483237160.png)

**Additional context**
my desktop is pantheon

**Notify maintainers**
@schmittlauch

**Metadata**
 - system: `"x86_64-linux"`
 - host os: `Linux 5.6.14, NixOS, 20.03.1950.48723f48ab9 (Markhor)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.4`
 - channels(root): `"home-manager-20.03, nixos-20.03.1950.48723f48ab9, nixos-hardware, nixos-unstable-20.09pre226148.0f5ce2fac0c"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: ktouch
# a list of nixos modules affected by the problem
module:
```

**Comment by schmittlauch** [comment]
@davidak Running under Plasma, I cannot reproduce this. But as I also do not get these TypeErrors, this might be a case of QtQuick impurities having slipped in again. (@ttuegel ?)

Let's see whether I find the time to craft together a Pantheon test environment for this.

**Comment by schmittlauch** [comment]
@davidak You might wanna try https://github.com/NixOS/nixpkgs/issues/79713#issuecomment-628775987

**Comment by davidak** [comment]
It should be in stable now, but the problem is not solved.

![Screenshot from 2020-09-01 16-07-04](https://user-images.githubusercontent.com/91113/91876348-e2006d00-ec7c-11ea-865a-e71d76b67cc0.png)

---

 - system: `"x86_64-linux"`
 - host os: `Linux 5.7.19, NixOS, 20.03.2868.ff6a070b4ef (Markhor)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.6`
 - channels(root): `"home-manager-20.03, nixos-20.03.2868.ff6a070b4ef, nixos-hardware, nixos-unstable-20.09pre239318.c59ea8b8a0e"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

**Comment by ttuegel** [comment]
> this might be a case of QtQuick impurities having slipped in again

Unfortunately, it's not "slipping in", Qt is impure by design. There's nothing I can do to make it "pure" in the Nix sense without breaking the core functionality.

@schmittlauch If you can't reproduce it on Plasma, I suspect it is a missing dependency, which happens to be installed under Plasma.
